client/workflow.go

```diff
@@ -0,0 +1,272 @@
+	}
+	if req.WorkflowComponent == "" {
+		req.WorkflowComponent = DefaultWorkflowComponent
+	}
+	_, err := c.protoClient.TerminateWorkflowBeta1(ctx, &pb.TerminateWorkflowRequest{
+		InstanceId:        req.InstanceID,
+		WorkflowComponent: req.WorkflowComponent,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to terminate workflow: %v", err)
+	}
+	return nil
+}
+
+// PauseWorkflowBeta1 pauses a workflow that can be resumed later using the beta1 spec.
+func (c *GRPCClient) PauseWorkflowBeta1(ctx context.Context, req *PauseWorkflowRequest) error {
+	if req.InstanceID == "" {
+		return errors.New("failed to pause workflow: InstanceID must be supplied")
+	}
+	if req.WorkflowComponent == "" {
+		req.WorkflowComponent = DefaultWorkflowComponent
+	}
+	_, err := c.protoClient.PauseWorkflowBeta1(ctx, &pb.PauseWorkflowRequest{
+		InstanceId:        req.InstanceID,
+		WorkflowComponent: req.WorkflowComponent,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to pause workflow: %v", err)
+	}
+	return nil
+}
+
+// ResumeWorkflowBeta1 resumes a paused workflow using the beta1 spec.
+func (c *GRPCClient) ResumeWorkflowBeta1(ctx context.Context, req *ResumeWorkflowRequest) error {
+	if req.InstanceID == "" {
+		return errors.New("failed to resume workflow: InstanceID must be supplied")
+	}
+	if req.WorkflowComponent == "" {
+		req.WorkflowComponent = DefaultWorkflowComponent
+	}
+	_, err := c.protoClient.ResumeWorkflowBeta1(c.withAuthToken(ctx), &pb.ResumeWorkflowRequest{
+		InstanceId:        req.InstanceID,
+		WorkflowComponent: req.WorkflowComponent,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to resume workflow: %v", err)
+	}
+	return nil
+}
+
+// RaiseEventWorkflowBeta1 raises an event on a workflow using the beta1 spec.
+func (c *GRPCClient) RaiseEventWorkflowBeta1(ctx context.Context, req *RaiseEventWorkflowRequest) error {
+	if req.InstanceID == "" {
+		return errors.New("failed to raise event on workflow: InstanceID must be supplied")
+	}
+	if req.WorkflowComponent == "" {
+		req.WorkflowComponent = DefaultWorkflowComponent
+	}
+	if req.EventName == "" {
+		return errors.New("failed to raise event on workflow: EventName must be supplied")
+	}
+	var eventData []byte
+	var err error
+	if req.SendRawData {
+		eventData = req.EventData.([]byte)
+	} else {
+		eventData, err = marshalInput(req.EventData)
+		if err != nil {
+			return fmt.Errorf("failed to raise an event on workflow: %v", err)
+		}
+	}
+
+	_, err = c.protoClient.RaiseEventWorkflowBeta1(c.withAuthToken(ctx), &pb.RaiseEventWorkflowRequest{
+		InstanceId:        req.InstanceID,
+		WorkflowComponent: req.WorkflowComponent,
+		EventName:         req.EventName,
+		EventData:         eventData,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to raise event on workflow: %v", err)
+	}
+	return nil
+}
+
+func marshalInput(input any) (data []byte, err error) {
+	if input == nil {
+		return nil, nil
+	}
+	if _, typeByteArray := input.([]byte); typeByteArray {
+		return input.([]byte), nil
```

Comment:
This feels a bit problematic. We already have `SendRawData` property which is supposed to handle this case, making this redundant. But also, it prevents workflow authors from accepting a JSON-serialized byte array. It's a corner case, but represents a behavior that's different from other SDKs and would be hard to debug. I recommend removing this check and always calling `json.Marshal` for non-nil, non-raw inputs, including byte arrays.